pkg/kubelet/pluginmanager/pluginwatcher/README.md

```diff
@@ -16,6 +16,56 @@ there.
 
 This socket filename should not start with a '.' as it will be ignored.
 
+To avoid conflicts between different plugins, the recommendation is to use
+`<plugin name>[-<some optional string>].sock` as filename. `<plugin name>`
+should end with a DNS domain that is unique for the plugin.
+
+If a plugin supports seamless upgrades, each instance should use a unique
+filename. Otherwise the following could happen:
```

Comment:
cc @bart0sh @linxiulei @ffromani @SergeyKanzhelev 

This is my conclusion from the discussions in https://github.com/kubernetes/kubernetes/pull/127821. Agreed?

Does the rest of the documentation update match your understanding of the current code base?